runtime/lib/capi/RuntimeCAPI.cpp

```diff
@@ -580,6 +580,20 @@ void __quantum__qis__MultiRZ(double theta, bool adjoint, int64_t numQubits, ...)
                                                              /* inverse = */ adjoint);
 }
 
+void __quantum__qis__ISWAP(QUBIT *wire0, QUBIT *wire1, bool adjoint)
+{
+    Catalyst::Runtime::getQuantumDevicePtr()->NamedOperation(
+        "ISWAP", {}, {reinterpret_cast<QubitIdType>(wire0), reinterpret_cast<QubitIdType>(wire1)},
+        adjoint);
+}
+
+void __quantum__qis__PSWAP(double phi, QUBIT *wire0, QUBIT *wire1, bool adjoint)
+{
+    Catalyst::Runtime::getQuantumDevicePtr()->NamedOperation(
+        "PSWAP", {phi},
```

Comment:
Hey @paul0403! Thanks for looking into this! Except classifying quantum gates, I can think of two feasible options: 1) Each device can have it's own list of supported gates. 2) Rely on backend devices APIs to check supported gates.

We have followed the second approach for all of our devices. But unfortunately, Lightning simulators are not able to raise meaningful error messages anymore! I will ensure that this will be fixed in the near future. For the scope of this PR, I think it should be enough to add a `RT_FAIL_IF` to the beginning of `NamedOperation` to raise an error if users call this method with `ISWAP` or `PSWAP` for now. You can update your C++ test to check this patch using `REQUIRE_THROWS_WITH`. 

Note that these gates are decomposed in the frontend for Lightning simulators, so there should not be any issues for users to use these gates from the frontend.

Reply:
@paul0403 let's remove this C++ test and all the `PSWAP` and `ISWAP` patches to `LightningSimulator` and `LightningKokkosSimulator` in this PR. The issue is related to the Lightning C++ API in https://github.com/PennyLaneAI/pennylane-lightning that we will provide a fix for it later. So, just ensure we have C++ test for the OpenQasm device. Thank you for digging into this issue! :rocket:

Reply:
@maliasadi Will do. Is there a way to indicate in `runtime/tests/Test_LightningGateSet.cpp` that a test should only be run on openqasm device?

Reply:
`Test_LightningGateSet.cpp` is a test suite for Lightning simulators tests. you can add your openqasm unit tests to `Test_OpenQasmDevice.cpp` and `Test_OpenQasmBuilder` to test on OpenQasm devices.

Based on the latest version of your work, you only need to add something like the following test to `Test_OpenQasmDevice.cpp`:

``` c++
TEST_CASE("Test PSWAP and ISWAP with BuilderType::Braket", "[openqasm]")
{
    constexpr size_t shots{1000};
    std::unique_ptr<OpenQasmDevice> device = std::make_unique<OpenQasmDevice>(
        "{device_type : braket.local.qubit, backend : default, shots : 1000}");

    constexpr size_t n{2};
    auto wires = device->AllocateQubits(n);

    device->NamedOperation("Hadamard", {}, {wires[0]}, false);
    device->NamedOperation("ISWAP", {}, {wires[0], wires[1]}, false);
    device->NamedOperation("PSWAP", {0.34}, {wires[0], wires[1]}, false);

    auto obs = device->Observable(ObsId::PauliZ, {}, std::vector<QubitIdType>{1});
    auto expval = device->Expval(obs);
    CHECK(expval == Approx(1).margin(1e-5));
}
```